File: LeetCode/Greedy/Golang/Leetcode2178-maximum-split-of-positive-even-integers.go
func maximumEvenSplit(finalSum int64) []int64 {
	if finalSum & 1 != 0 {
			return []int64{}
	}
	ans := []int64{}
	for i := int64(2);finalSum >= i;i += 2 {
			ans = append(ans, i)
			finalSum -= i
	}
	if finalSum != 0 {
			n := len(ans)
			ans[n - 1] += finalSum
	}
	return ans
}

File: LeetCode/Greedy/JS/Leetcode2178-maximum-split-of-positive-even-integers.js
var maximumEvenSplit = function (finalSum) {
  if (finalSum & 1) {
    return [];
  }
  let ans = [];
  for (let i = 2; finalSum >= i; i += 2) {
    ans.push(i);
    finalSum -= i;
  }
  if (finalSum) {
    const n = ans.length;
    ans[n - 1] += finalSum;
  }
  return ans;
};